foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/HiveUniversalConfigYAMLMigration.java

```diff
@@ -0,0 +1,110 @@
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import com.boozallen.aissemble.upgrade.util.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.error.YAMLException;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * This migration removes hive username in the hive-metastore-service values.yaml to use values from configuration store service only if values.yaml has default value.
+ */
+public class HiveUniversalConfigYAMLMigration extends AbstractAissembleMigration {
+    private static final Logger logger = LoggerFactory.getLogger(HiveUniversalConfigYAMLMigration.class);
+    private static final List<String> propertiesForConfigStore = Arrays.asList("javax.jdo.option.ConnectionUserName");
+    private static final List<String> mysqlAuthForConfigStore = Arrays.asList("username");
+    private static final HashSet<String> linesToRemove = new HashSet<>(Arrays.asList("username: hive", "- name: javax.jdo.option.ConnectionUserName" ,"value: hive", "description: Username to use against metastore database", "# NB: UCS"));
+    private static final String DEFAULT_HIVE_USERNAME_AND_PASSWORD = "hive";
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try {
+            YamlUtils.YamlObject yaml = YamlUtils.loadYaml(file);
+            if(yaml.hasObject("aissemble-hive-metastore-service-chart"))
+            {
+                yaml = yaml.getObject("aissemble-hive-metastore-service-chart");
+            }else{
+                return false;
+            }
+            if (yaml.hasObject("mysql", "auth")) {
+                YamlUtils.YamlObject mysqlAuth = yaml.getObject("mysql", "auth");
+                for(String key : mysqlAuth.keySet())
+                {
+                    if(mysqlAuthForConfigStore.contains(key) && mysqlAuth.get(key).equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                    {
+                        return true;
+                    }
+                }
+
+            }
+
+            if (yaml.hasList("configMap", "metastoreServiceConfig", "properties")) {
+                List<YamlUtils.YamlObject> properties = yaml.getListOfObjects("configMap", "metastoreServiceConfig", "properties");
+                for(YamlUtils.YamlObject property : properties)
+                {
+                    if(propertiesForConfigStore.contains(property.get("name")) && property.get("value").equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                    {
+                        return true;
+                    }
+                   }
+            }
```

Comment:
Think we want both of these conditions to be true in order to run the migration. So the return could be something like:
```
return hasMySqlAuthDefaultUsername && hasMetastoreServiceConfigDefaultUsername
```
and you set those booleans in each of the inner most if blocks